### PR TITLE
Blog: Rails RESTful api outside in tdd

### PR DESCRIPTION
**Pull request by nitinrajkumar451**
The following is a blog about a , conceited programmer's journey in developing a simple TODO RESTful API using Ruby on Rails. How the quest for solutions to the challenges he faced at every step drifted him towards employing outside in Test driven development. Written in first person subjective.

### CODE REVIEW

**Comment by 2KAbhishek** [comment]
Just made a bunch of fixes to the content.

I'd suggest running it through grammarly once.
After that open up the preview on your local with `hugo serve`, some sections look too dense and wordy, trim it or add some separation to improve readability.

**Review by 2KAbhishek — CHANGES_REQUESTED** [review]
Please check comment above

**Review by 2KAbhishek — APPROVED** [review]

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+           expected: 1 time with arguments: ({:description=>"First todo", :title=>"Todo-1"})
+           received: 0 times
+
+GREEN:
+    def create_todo(add_params)
+        @repo.new_todo(add_params)
+    end
+```
+
+Now that the runner spec is fixed, so has run the request spec to check if it has been fixed too?
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:539]: Need more clarity.

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+            return error ,:unprocessable_entity
+        end
+    end
+ $ rspec
+
+All Green
+
+```
+
+### It is always quite a sight to see all green
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:732]: ### All Green - quite a wonderful sight!

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+
+I was so immersed into conversing with rspec to develop this code, I felt like I was a part of a play, playing a character and that the play is being observed, and I am responding only when there is a cue and refrained to when there is no cue as I may mislead them observers if I take any decision without the cue from the rspec.
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:734]: I was so immersed in conversing with rspec while developing this application that I kind of got distanced from reality. I felt like I’d embodied this character in a play that was being watched and I was to only act out my lines on cue lest I’d mislead the audience. Rspec called the shots here!

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 4]
```diff
@@ -0,0 +1,740 @@
...
+
+In this manner I finished the assignment and gave it to the client.
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:736]: And there you have it, folks! I finally ended up finishing the assignment and got it ready to be handed over to the client!

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 5]
```diff
@@ -0,0 +1,740 @@
...
+I had two days to go for a visit around the place.
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:737]: With a couple of days left of my stay, I started out on my journey exploring the unknown!

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 6]
```diff
@@ -0,0 +1,740 @@
...
+As I was watching a tomb my instinctive thoughts about the plausible errors in my work started to take over me but with the fact that there never existed any piece of code before going through a test did not let those thoughts take over me, and they seem to have evaporated.
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:738]: At one such moment, I was looking at a tomb over at the distance when dreadful thoughts started to creep back in! Thoughts of possible errors in the app! But, I put all of that to rest as I’d let the tests do their magic now. 



------did I change the meaning of the block?

File: content/blog/rails-restapi-outside-in-tdd.md [hunk 7]
```diff
@@ -0,0 +1,740 @@
...
+The sightseeing was pleasant with nothing in the back of my mind.
```
Comment (yoursPerilously) [inline content/blog/rails-restapi-outside-in-tdd.md:739]: Now, with my mind at ease and feeling a sense of achievement, the sights all around seemed a bit more pleasant!

**Review by yoursPerilously — CHANGES_REQUESTED** [review]

**Review by 2KAbhishek — APPROVED** [review]

**Pull request merged**